response_operations_ui/views/collection_exercise.py

```diff
@@ -326,23 +326,40 @@ def get_create_collection_exercise_form(survey_ref, short_name):
     survey_details = survey_controllers.get_survey(short_name)
     return render_template('create-collection-exercise.html', form=form, short_name=short_name,
                            survey_ref=survey_ref, survey_id=survey_details['survey']['id'],
-                           survey_name=survey_details['collection_exercises'][0]['name'])
+                           survey_name=survey_details['survey']['shortName'])
 
 
 @collection_exercise_bp.route('/<survey_ref>-<short_name>/create-collection-exercise', methods=['POST'])
 @login_required
 def create_collection_exercise(survey_ref, short_name):
-    form = CreateCollectionExerciseDetailsForm(form=request.form)
-    if not form.validate():
+    ce_form = CreateCollectionExerciseDetailsForm(form=request.form)
+
+    survey_details = survey_controllers.get_survey(short_name)
```

Comment:
Do we need this call? We have short_name and the survey id as a hidden field in the form.